airbyte-integrations/connectors/source-mixpanel/source_mixpanel/source.py

```diff
@@ -804,7 +804,12 @@ def check_connection(self, logger: AirbyteLogger, config: Mapping[str, Any]) ->
         :param logger:  logger object
         :return Tuple[bool, any]: (True, None) if the input config can be used to connect to the API successfully, (False, error) otherwise.
         """
-        auth = TokenAuthenticatorBase64(token=config["api_secret"])
+        if (config["api_secret"] ) {
+            auth = TokenAuthenticatorBase64(token=config["api_secret"])
+        }
+        if (config["serviceaccount_username"] ) {
+            auth = TokenAuthenticatorBase64(token=config["serviceaccount_username"] + ":" + config["serviceaccount_secret"])
```

Comment:
Was this `TokenAuthenticatorBase64` class already imported?

Reply:
`TokenAuthenticatorBase64` is defined in the same file.